internal/layer/sbom_restorer.go

```diff
@@ -0,0 +1,131 @@
+package layer
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+
+	"github.com/buildpacks/imgutil"
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/buildpack"
+	io2 "github.com/buildpacks/lifecycle/internal/io"
+	"github.com/buildpacks/lifecycle/launch"
+	"github.com/buildpacks/lifecycle/layers"
+)
+
+//go:generate mockgen -package testmock -destination testmock/sbom_restorer.go github.com/buildpacks/lifecycle/internal/layer SBOMRestorer
+type SBOMRestorer interface {
+	RestoreFromPrevious(image imgutil.Image, layerDigest string) error
+	RestoreFromCache(cache Cache, layerDigest string) error
+	RestoreToBuildpackLayers(detectedBps []buildpack.GroupBuildpack) error
+}
+
+type Cache interface {
+	RetrieveLayer(sha string) (io.ReadCloser, error)
+}
+
+func NewSBOMRestorer(layersDir string, logger Logger) SBOMRestorer {
+	return &DefaultSBOMRestorer{
+		layersDir: layersDir,
+		logger:    logger,
+	}
+}
+
+type DefaultSBOMRestorer struct {
+	layersDir string
+	logger    Logger
+}
+
+func (r *DefaultSBOMRestorer) RestoreFromPrevious(image imgutil.Image, layerDigest string) error {
+	// Sanity check to prevent panic.
+	if image == nil {
+		return errors.Errorf("restoring layer: previous image not found for %q", layerDigest)
+	}
+	r.logger.Debugf("Retrieving previous image layer for %q", layerDigest)
```

Comment:
I might suggest changing this so we can tell it from the other restore method.
```suggestion
	r.logger.Debugf("Retrieving previous image layer for sbom layer %q", layerDigest)
```

Reply:
Added in fbbb2e66